packages/edit-site/src/components/navigation-sidebar/navigation-panel/menus/templates.js

```diff
@@ -11,41 +11,121 @@ import {
 	__experimentalNavigationMenu as NavigationMenu,
 } from '@wordpress/components';
 import { useSelect } from '@wordpress/data';
-import { __, _x } from '@wordpress/i18n';
-import { useState, useCallback } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+import { useState, useCallback, useMemo } from '@wordpress/element';
+import { store as coreStore } from '@wordpress/core-data';
 
 /**
  * Internal dependencies
  */
-import TemplatesPagesMenu from './templates-pages';
-import TemplatesPostsMenu from './templates-posts';
 import {
 	MENU_ROOT,
 	MENU_TEMPLATES,
-	MENU_TEMPLATES_ALL,
+	MENU_TEMPLATES_GENERAL,
 	MENU_TEMPLATES_PAGES,
 	MENU_TEMPLATES_POSTS,
-	TEMPLATES_GENERAL,
+	MENU_TEMPLATES_UNUSED,
+	TEMPLATES_TOP_LEVEL,
 } from '../constants';
-import TemplatesAllMenu from './templates-all';
 import NewTemplateDropdown from '../new-template-dropdown';
 import TemplateNavigationItem from '../template-navigation-item';
 import SearchResults from '../search-results';
+import TemplatesSubMenu from './templates-sub';
+import { isTemplateSuperseded } from '../template-hierarchy';
+
+function getTemplateLocation( template ) {
+	const { slug } = template;
+
+	if ( TEMPLATES_TOP_LEVEL.includes( slug ) ) {
+		return MENU_TEMPLATES;
+	}
+
+	if ( slug === 'page-home' ) {
+		return MENU_TEMPLATES_GENERAL;
+	}
+
+	if (
+		slug.startsWith( 'post-' ) ||
```

Comment:
Maybe we can isolate these to an array similar to `TEMPLATES_TOP_LEVEL`?

Reply:
Good idea! Thanks!